onnxscript/tests/models/pnp.py

```diff
@@ -0,0 +1,262 @@
+    grid_w = grid_w_0 + zeros_D_H_W
+
+    grid_h_0 = op.Range(0, H * step_H, step_H)
+    zeros_D_W_H = op.Transpose(zeros_D_H_W, perm=[0, 2, 1])
+    grid_h_1 = zeros_D_W_H + grid_h_0
+    grid_h = op.Transpose(grid_h_1, perm=[0, 2, 1])
+
+    grid_d_0 = op.Range(0, D * step_D, step_D)
+    zeros_H_W_D = op.Transpose(zeros_D_H_W, perm=[1, 2, 0])
+    grid_d_1 = zeros_H_W_D + grid_d_0
+    grid_d = op.Transpose(grid_d_1, perm=[2, 0, 1])
+                        
+    original_grid_start_seq = op.SequenceConstruct(grid_d, grid_h, grid_w)
+    original_grid_start_stack = op.ConcatFromSequence(original_grid_start_seq, axis=-1, new_axis=1)  # [D, H, W, 3]
+    original_grid_start = op.Reshape(original_grid_start_stack, op.Constant(value_ints=[-1, 3]))  # [D*H*W, 3]
+    original_grid_start = op.Min(original_grid_start, image_size - patch_size)
+    original_grid_stop = original_grid_start + patch_size
+    original_grid_seq = op.SequenceConstruct(original_grid_start, original_grid_stop)
+    slices = op.ConcatFromSequence(original_grid_seq, axis=-1, new_axis=1)  # [D*H*W, 3, 2]
+    return slices
+
+@script()
+def prepare_for_predictor_batch_size_is_1_script(inputs: FLOAT["N", 1, "D", "H", "W"], slice_g: INT64, slices: INT64["N", 3, 2]) -> (FLOAT["N", 1, "roi_D", "roi_H", "roi_W"], INT64[3], INT64[3]):
+    # assume batch size N is 1
+    num_win, _, _ = op.Split(op.Shape(slices), num_outputs=3)
+    spatial_axes = op.Range(2, 5, 1)
+    zero_int = op.Constant(value_ints=[0])
+    one_int = op.Constant(value_ints=[1])
+    two_int = op.Constant(value_ints=[2])
+    slice_g_ = slice_g + zero_int
+    slice_g_1 = slice_g + one_int
+    start_ = op.Slice(slices, op.Concat(slice_g_, zero_int, axis=0), op.Concat(slice_g_1, one_int, axis=0), op.Constant(value_ints=[0, 2]))
+    start = op.Reshape(start_, op.Constant(value_ints=[-1]))
+    stop_ = op.Slice(slices, op.Concat(slice_g_, one_int, axis=0), op.Concat(slice_g_1, two_int, axis=0), op.Constant(value_ints=[0, 2]))
+    stop = op.Reshape(stop_, op.Constant(value_ints=[-1]))
+    win_data = op.Slice(inputs, start, stop, spatial_axes)
+    # if sw_batch_size > 1:
+    #     for idx in op.Range(slice_g + 1, op.Min(slice_g + sw_batch_size, num_win)):
+    #         win_data = op.Concat(win_data, op.Slice(inputs, slices[idx, :, 0], slices[idx, :, 1], spatial_axes))
+    return win_data, start, stop
+
+@script()
+def sliding_window_inference(inputs: FLOAT["N", 1, "D", "H", "W"], roi_size: INT64[3]) -> FLOAT["N", "Seg_C", "D", "H", "W"]:
+    """
+    for simplicity, we assume that the step size is the same as the roi size. D/H/W are multiple of roi_size in 3 dimensions,
+    no overlay, no padding. weight is 1.
+    TODOs: sw_batch_size > 1, roi_size is not multiple of D/H/W, overlay, padding, weight is not 1.
+    """
+
+    inputs_shape = op.Shape(inputs)
+    inputs_spatial_shape = op.Slice(inputs_shape, op.Constant(value_ints=[2]), op.Constant(value_ints=[5]), op.Constant(value_ints=[0]))

```

Comment:
nit: may be this will be more readable if we move it after the line below, and rewrite it to:
```py
   inputs_spatial_shape = op.Concat(D, H, W, axis=0)
```
Alternatively, we can also say:
```py
   inputs_spatial_shape = op.Shape(inputs, from=2)
```